src/scripts/main.js

```diff
@@ -1,5 +1,26 @@
 'use strict';
 
+const wall = document.querySelector('.wall');
+const spider = wall.querySelector('.spider');
+
+const topMargin = wall.offsetTop + wall.clientTop + spider.clientHeight / 2;
+const leftMargin = wall.offsetLeft + wall.clientLeft + spider.clientWidth / 2;
+
+const topMax = wall.clientHeight - spider.clientHeight;
+const leftMax = wall.clientWidth - spider.clientWidth;
+
 document.addEventListener('click', (e) => {
-  // write code here
+  if (e.target.closest('.wall')) {
```

Comment:
The use of `e.target.closest('.wall')` is appropriate for checking if the click event occurred within the wall. However, ensure that the `.wall` class is not applied to any nested elements within the wall, as this might lead to unexpected behavior.